plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -1050,13 +1062,43 @@ coverageCompile originalExpr exprType src compiledTerm covT =
     findHeadSymbol (GHC.Cast t _) = findHeadSymbol t
     findHeadSymbol _              = Nothing
 
+-- | We cannot compile the unfolding of `GHC.Num.Integer.integerNegate`, which is important because GHC inserts calls to it when it sees negations, even negations of literals (unless NegativeLiterals is on, which it usually isn't). So we directly
+-- define a PIR term for it: @integerNegate = \x -> 0 - x@.
+defineIntegerNegate :: (CompilingDefault PLC.DefaultUni fun m ann) => m ()
+defineIntegerNegate = do
+  ghcId <- GHC.tyThingId <$> getThing 'GHC.Num.Integer.integerNegate
+  -- Always inline `integerNegate`.
```

Comment:
why?

Reply:
Otherwise we get a lot of `integerNegate`s in _optimized_ PIR!

Reply:
Right, but is that bad?

Reply:
> And without the annMayInline annotation, the inliner cannot make this transformation.

not today! Maybe we should fix that! 

I just want you to write down the reasoning, it's not obvious.

Reply:
Added:
> -- Always inline `integerNegate`.
  -- `let integerNegate = \x -> 0 - x in integerNegate 1 + integerNegate 2`
  -- is much more expensive than `(-1) + (-2)`. The inliner cannot currently
  -- make this transformation without `annAlwaysInline`, because it is not aware
  -- of constant folding.